lib/quirc.c

```diff
@@ -44,11 +48,40 @@ void quirc_destroy(struct quirc *q)
 	free(q);
 }
 
+int quirc_init(struct quirc *q, int w, int h, uint8_t *image)
+{
+	if (!q || !image || !QUIRC_PIXEL_ALIAS_IMAGE)
+		return -1;
+
+	/*
+	 * Same sanity check as in quirc_resize() for the same reasons.
+	 */
+	if (w < 0 || h < 0)
+		return -1;
+
+	memset(q, 0, sizeof(*q));
+	q->w = w;
+	q->h = h;
+	q->image = image;
+	return 0;
+}
+
 int quirc_resize(struct quirc *q, int w, int h)
 {
 	uint8_t		*image  = NULL;
 	quirc_pixel_t	*pixels = NULL;
 
+	if (!q->need_to_free) {
```

Comment:
IMHO this codepath is an example showing that `need_to_free` is not an ideal member name.

Reply:
I don't follow.  What is your recommendation?

Reply:
I think either `q->malloc` or `q->dynamic` would be more readable.